### ISSUE TITLE
Docker containers aren't sufficiently tested

### ISSUE THREAD

**Issue by maxalbert** [Flowminder/FlowKit#683]
**Describe the bug**
On a fresh installation FlowAPI fails to start up when running `make up`.

**Product**
Flowapi

**Version**
Current master (93f5f1e7bf58d6e20b9a38e8ffe0c32f154e14ed).

**To Reproduce**
On a fresh VM:
```
$ git clone https://github.com/Flowminder/FlowKit.git
$ cd FlowKit
$ set -a
$ source development_environment
$ set +a
$ make up
```
Wait for a while until all the images are built and the containers are running.

Then check the status and see that `flowapi` failed to start up:
```
$ docker ps
CONTAINER ID        IMAGE                               COMMAND                  CREATED             STATUS                            PORTS                           NAMES
34bc5fba4d07        flowminder/flowmachine:latest       "pipenv run flowmach…"   3 minutes ago       Up 2 minutes                      0.0.0.0:5555->5555/tcp          flowmachine
296e4ccce4a2        flowminder/flowdb-testdata:latest   "docker-entrypoint.s…"   3 minutes ago       Up 3 minutes                      0.0.0.0:9000->5432/tcp          flowdb_testdata
2f33e094adb0        bitnami/redis                       "/entrypoint.sh /run…"   3 minutes ago       Up 3 minutes                      0.0.0.0:6379->6379/tcp          redis_flowkit
77cfd16a5100        flowminder/flowapi:latest           "pipenv run ./start.…"   3 minutes ago       Restarting (127) 55 seconds ago                                   flowapi
7f3094030d71        flowminder/flowauth:latest          "/entrypoint.sh /sta…"   3 minutes ago       Up 3 minutes                      443/tcp, 0.0.0.0:9091->80/tcp   flowauth
```

**Expected behavior**
All containers (including flowapi) should start up successfully.

**Additional context**
The flowapi docker logs say:
```
$ docker logs flowapi 
env: can't execute 'bash': No such file or directory
env: can't execute 'bash': No such file or directory
env: can't execute 'bash': No such file or directory
[...]
```
I assume the error stems from the shebang line in `flowapi/start.sh`:
```
#!/usr/bin/env bash
```
So the solution would be to either install bash or run the script through `sh` instead of `bash`.

However, I'm surprised that this isn't caught by any of the CI tests.

**Comment by maxalbert** [comment]
Note that before fixing this issue we should also ensure that at least one of the CI tests fails with the current setup, to ensure that this doesn't accidentally slip through again in the future.

**Comment by greenape** [comment]
Oh bugger. That’ll be the switch to alpine!

**Comment by greenape** [comment]
I think a ping job would probably suffice. Suppose we kind of want this for all the containers really.

**Comment by maxalbert** [comment]
Shouldn't the integration tests catch this though?

**Comment by greenape** [comment]
Or running the integration tests twice - once for coverage, once to check the containers?

**Comment by maxalbert** [comment]
Why wouldn't simply running the integration tests once detect that flowapi isn't running?

**Comment by greenape** [comment]
Integration tests currently run them outside the containers in order to collect coverage data

**Comment by greenape** [comment]
(The immediate issue was closed by #685, but keeping this open until we have the testing in place to stop it happening again.)

**Comment by maxalbert** [comment]
Yes, that seems sufficient for this issue. 👍
